bucketchecker.go

```diff
@@ -0,0 +1,7 @@
+package gocbcorex
+
+import "context"
+
+type BucketChecker interface {
```

Comment:
We have BucketChecker vs ManifestFetcher, should probably make these consistent?

Reply:
I guess they are subtly different in behaviour.